integration/volume/volume_test.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/docker/docker/api/types/filters"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/request"
+	"github.com/docker/docker/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVolumesList(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prefix, _ := getPrefixAndSlashFromDaemonPlatform()
+
+	id := container.Run(t, ctx, client, container.WithVolume(prefix+"foo"))
+
+	c, err := client.ContainerInspect(ctx, id)
+	require.NoError(t, err)
+	vname := c.Mounts[0].Name
+
+	names := func(volumes []*types.Volume) []string {
+		entries := []string{}
+		for _, vol := range volumes {
+			entries = append(entries, vol.Name)
+		}
+		return entries
+	}
+	volumes, err := client.VolumeList(ctx, filters.Args{})
+	require.NoError(t, err)
+	assert.Contains(t, names(volumes.Volumes), vname)
+}
+
+func TestVolumesCreate(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	name := "test"
+	vol, err := client.VolumeCreate(ctx, volumetypes.VolumesCreateBody{
+		Name: name,
+	})
+	require.NoError(t, err)
+
+	assert.Equal(t, filepath.Base(filepath.Dir(vol.Mountpoint)), name)
+}
+
+func TestVolumesRemove(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prefix, _ := getPrefixAndSlashFromDaemonPlatform()
+
+	name := "test"
+	id := container.Run(t, ctx, client, container.WithName(name), container.WithVolume(prefix+"foo"))
+
+	c, err := client.ContainerInspect(ctx, id)
+	require.NoError(t, err)
+	vname := c.Mounts[0].Name
+
+	err = client.VolumeRemove(ctx, vname, false)
+	testutil.ErrorContains(t, err, "volume is in use")
+
+	err = client.ContainerRemove(ctx, name, types.ContainerRemoveOptions{
+		Force: true,
+	})
+	require.NoError(t, err)
+
+	err = client.VolumeRemove(ctx, vname, false)
+	require.NoError(t, err)
+}
+
+func TestVolumesInspect(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	// sampling current time minus a minute so to now have false positive in case of delays
+	now := time.Now().Truncate(time.Minute)
+
+	name := "test"
+	_, err := client.VolumeCreate(ctx, volumetypes.VolumesCreateBody{
+		Name: name,
+	})
+	require.NoError(t, err)
+
+	vol, err := client.VolumeInspect(ctx, name)
+	require.NoError(t, err)
+	assert.Equal(t, vol.Name, name)
+
+	// comparing CreatedAt field time for the new volume to now. Removing a minute from both to avoid false positive
+	testCreatedAt, err := time.Parse(time.RFC3339, strings.TrimSpace(vol.CreatedAt))
+	require.NoError(t, err)
+	testCreatedAt = testCreatedAt.Truncate(time.Minute)
+	assert.Equal(t, testCreatedAt.Equal(now), true, "Time Volume is CreatedAt not equal to current time")
```

Comment:
https://godoc.org/github.com/gotestyourself/gotestyourself/assert/cmp#DeepEqual should make this type of comparison a lot cleaner. I hope we can use it soon.

Reply:
@dnephin Thanks. It seems that a vendor update is needed. I will take a look and create a follow up PR for that.

Reply:
Don't worry about using it just yet. I have an automated migration that we can use to convert things all at once. I'll open a PR for it soon.